From 5a6cd042d3cee9e3bbf3b0125a89e64d75dffdc7 Mon Sep 17 00:00:00 2001
From: Cristian Stoica <cristian.stoica@freescale.com>
Date: Thu, 31 Oct 2013 13:13:51 +0200
Subject: [[Patch][fsl 04/16] fix scatterlist size for in-place operations

Upstream-status: Pending

For in-place operations, the scatterlist must be big enough to allow for
either the source or destination data:
 - for block ciphers, the destination will be bigger than the source if
   the source is not multiple of blocksize
 - for aead ciphers, the additional data and padding guarantees the
   destination to be bigger than the source.

Signed-off-by: Cristian Stoica <cristian.stoica@freescale.com>
Tested-by: Horia Ioan Geanta Neag <horia.geanta@freescale.com>
---
 zc.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/zc.c b/zc.c
index 809a36d..8855083 100644
--- a/zc.c
+++ b/zc.c
@@ -162,8 +162,9 @@ int get_userbuf(struct csession *ses,
 	}
 
 	if (src == dst) {	/* inplace operation */
-		rc = __get_userbuf(src, src_len, 1, ses->used_pages,
-			               ses->pages, ses->sg, task, mm);
+		rc = __get_userbuf(src, max(src_len, dst_len), 1, ses->used_pages,
+				ses->pages, ses->sg, task, mm);
+
 		if (unlikely(rc)) {
 			derr(1, "failed to get user pages for data IO");
 			return rc;
-- 
1.7.9.7

